sympy/algebras/quaternion.py

```diff
@@ -321,13 +321,25 @@ def pow(self, p):
         if p == -1:
             return q.inverse()
         res = 1
+
         while p > 0:
             if p & 1:
                 res = q * res
 
             p = p >> 1
             q = q * q
 
+        if p < -1:
```

Comment:
Will `q_inv = q.inverse()` and then returning `q_inv.power(-p)` work?

Reply:
Maybe this could be before the loop above:

    if p < 0:
        q, p = q.inverse(), -p

Reply:
Yes, ofcourse. That will work better.